### PR TITLE
upgrade msbuild locator to support msbuild 16

### LINKED ISSUE #4437: Support MSBuild 16 (VS 2019)
Now users with VS 2019 alone cannot generate site correctly. https://github.com/dotnet/docfx/issues/2562#issuecomment-482229787 

Work items:

- [x] Upgrade from `net462` to `net472`
- [ ] Upgrade `Microsoft.Build` to 16
- [ ] Updrade logic to locate `msbuild`

### PR DESCRIPTION
**Pull request by superyyrrzz**
resolve #4437
depend on #4616

Note that the PR check is expected to fail, unless it uses `vmImage: 'windows-2019'` specified in this PR.

### CODE REVIEW

File: Documentation/tutorial/howto_build_your_own_type_of_documentation_with_custom_plug-in.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Preparation
     otherwise add the nuget packages with the same version as DocFX.
 
 4.  Add framework assembly references:
-    `PresentationCore`, `PresentationFramework`, `WindowsBase`. (This step is optional in Visual Studio 2017)
```
Comment (herohua) [inline Documentation/tutorial/howto_build_your_own_type_of_documentation_with_custom_plug-in.md]: shall we just say 2019?

**Comment by SamuelEnglard** [comment]
Any idea when this will be released?

**Comment by superyyrrzz** [comment]
@SamuelEnglard Should be this week. Target 6/12 currently.

**Comment by superyyrrzz** [comment]
@SamuelEnglard @bitbonk @christianbumann  Just find that this change will break Linux/Mono... Any idea?

message:
```
Workspace failed with: [Failure] Msbuild failed when processing the file '/home/docfx-seed/src/SampleClass1/CatLibrary/CatLibrary.csproj' with message: The imported project "/usr/lib/mono/msbuild/15.0/bin//Microsoft.CSharp.Core.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.  /usr/lib/mono/msbuild/15.0/bin/Microsoft.CSharp.CurrentVersion.targets
```

**Comment by christianbumann** [comment]
Maybe the wrong msbuild vesion? 15.0 instead of 16.0? Und

**Comment by KalleOlaviNiemitalo** [comment]
@superyyrrzz, sorry, not familiar with running MSBuild on Mono.

**Pull request merged**